### PR TITLE
feat: uninstall command

### PR DESCRIPTION
**Pull request by potato4d**
## About

I added a new uninstall command.
This will do `npm uninstall` or `yarn remove`.

## Changes

- [x] Implement uninstall to new surface API
- [x] Add new surface API description to document
- [x] Add type definition to uninstall command
- [x] Update and write tests

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -73,6 +73,31 @@ export default class LMIFY {
...
+    if (!Array.isArray(packages)) {
+      packages = [packages]
+    }
+
+    if (!packages.length) {
+      return
+    }
+
+    const [grant] = await Promise.all([
+      this._grant(packages),
```
Comment (pi0) [inline src/index.js:91]: Maybe we can pass `{ action: 'uninstall' }` as the second argument to grantees?

File: src/index.js [hunk 2]
```diff
@@ -73,6 +73,31 @@ export default class LMIFY {
...
+      this.init()
+    ])
+
+    if (!grant) {
+      return
+    }
+
+    return this.packageManager.uninstall(packages)
```
Comment (pi0) [inline src/index.js:99]: `install`/`uninstall` logic seems duplicate now. What do you think if we add `do(action, ...args)` to the `package-manager` and also main class? `uninstall(packages, options` calls `this.do('uninstall', ...args)`

Comment (pi0) [reply src/index.js:99]: (useful for later actions like upgrade, verify)

Comment (potato4d) [reply src/index.js:99]: I don't always want to do the same thing with install and uninstall.
Implementing do in package-managers might be a good idea, but applying it to the main also seems to be blocking future extensibility.
For example, how about providing install / uninstall as a Surface API and preparing (private) `_do()` internally?

Comment (pi0) [reply src/index.js:99]: What do you think would be different for `install` and `uninstall`? For wrapper LMIFY class, it is all normalizing packages and ensuring class is in the ready state. Other options are abstracted as 3rd `options` param

**Comment by pi0** [comment]
This project is being deprecated. I suggest you try [unjs/nypm](https://github.com/unjs/nypm) instead.

**Pull request closed without merging**